pytorch_lightning/trainer/training_loop.py

```diff
@@ -305,6 +306,11 @@ def process_output(self, output, train):
         pass
 
     def train(self):
+
+        # Train begin callbacks
+        for callback in self.callbacks:
+            callback.on_train_begin()
```

Comment:
I would suggest to add something like `Trainer` hooks. So we can just call `self.on_train_begin()` in this place, and move
```
for callback in self.callbacks:
    callback.on_train_begin()
```
into `trainer.on_train_begin()`. I think it would be more clear because actually there are other things which should be called at certain points in training (for example, model hooks)

Reply:
Agree. The logic now lives in `TrainerCallbackHookMixin`.